### ISSUE TITLE
about Epoch

### ISSUE THREAD

**Issue by dogyoonlee** [thomasneff/AdaNeRF#7]
Hi! I appreciate to your awesome work.

However, I have a question about the epoch since you create a new data loader.

Does `epoch` in your code mean the same thing as `iteration` in traditional nerf code?
For example, 300000 `epoch` in your code equals 300000 `iteration` in traditional nerf code?

Or is it 1 `epoch` to go around the entire dataset as in traditional deep learning code?

**Comment by thomasneff** [comment]
Hi! 

You're exactly right, I think this might have been a misnomer that carried over from previous' code bases that we never got around to fixing. Epoch is the same as an interation and not really an epoch, yeah.

**Comment by dogyoonlee** [comment]
@thomasneff 
Thank you for your kind and quick reply!!

In addition, I have some more questions about the lossweight to train the sample network and nerf.

In your paper, you said that you initialize the sample network to make initial output to be zero and set the initial loss weight for each loss(mse loss and loss that makes initial sample importance to be 1) as 0.01 and 1.0 .
However, I cannot find that part in your code.
Loss weights are samely applied across the entire training as 1.0 and 0.0025 to mse and sparsity loss, except iterations before 25000 to apply the sparsity loss after dense training.

In addition, nerf network is locked before 1000 and sample network is locked after 80000 iters, which is not fully explained in your paper.
could you explain about this setting?

It will be really helpful to implement your code in my environment, which is purely coded in pure puthon and pytorch without cuda!

I'm really appreciate to your kind reply again!

**Comment by thomasneff** [comment]
Hi!

The corresponding losses are implemented here:

https://github.com/thomasneff/AdaNeRF/blob/main/src/losses.py#L87

Essentially, we start with the `NeRFWeightMultiplicationLoss` forcing initial outputs to be one (not zero) because `lossBlendingStart ` is at 25000. This means before iteration 25000, we only have the "One" loss in `NeRFWeightMultiplicationLoss`. Afterwards, we slowly blend from "One" to "Zero" and also add in the "NerfA" that computes an L1 loss w.r.t. the alpha outputs. The logic of  blending between the losses all happens within `NeRFWeightMultiplicationLoss`.

Note that our configs have `0.0025` as a baseline value, and we show the exact loss weights we used for each scene in our supplementary material (Table 1):

https://thomasneff.github.io/adanerf/adanerf_supplementary.pdf

About the weight locking:

We mention that the sampling network is locked throughout the sparse training phase in the paper, and that the sparse training phase is done for around ~220k iterations, that's where the 80000 in the config comes from, as we lock if after 80000.

The shading network being locked for 1000 iteration at the start indeed seems to be missing from the paper, but this is just to ensure that the sampling network is actually outputting constant "One"s before we start with dense training - this likely is not super impactful and is only a minor detail.

Hope this explains your questions!

**Comment by dogyoonlee** [comment]
@thomasneff 
Thank you for your kind reply!!
It really helpful to me to implement your idea!

by the way, do you still on the research field related to accelerate the nerf training or efficient rendering using nerf?

**Comment by dogyoonlee** [comment]
@thomasneff 

I'm sorry to bother you again.

As I understand your code, dense training and fine training is executed separately when train the AdaNeRF.

Then, is it right that the learning rate to train the `fine-tuning` is reset to `1e-4` after `dense training + sparsification + sparse training`?
